src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Features/IServerAddressesFeature.cs

```diff
@@ -8,5 +8,7 @@ namespace Microsoft.AspNetCore.Hosting.Server.Features
     public interface IServerAddressesFeature
     {
         ICollection<string> Addresses { get; }
+
+        bool PreferHostingUrls { get; set; }
```

Comment:
We discussed about this. The problem is that the feature is set by the server. Hosting sets it on the interface. Would have to cast to do that.